app/controllers/reviews_controller.rb

```diff
@@ -1,20 +1,53 @@
 class ReviewsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+
+  def index
+    @reviews = Review.all.limit(10)
+  end
+
   def new
+    @sitcom = Sitcom.find(params[:sitcom_id])
     @review = Review.new
   end
 
   def create
+    @sitcom = Sitcom.find(params[:sitcom_id])
     @review = current_user.reviews.new(review_params)
+    @review.sitcom_id = @sitcom.id
     if @review.save
       flash[:notice] = 'Successfully created your review.'
+      redirect_to sitcom_path(@sitcom)
     else
       flash[:alert] = @review.errors.full_messages
+      render :new
     end
-    redirect_to reviews_path
   end
 
-  def index
-    @reviews = Review.all.limit(10)
+  def edit
+    @sitcom = Sitcom.find(params[:sitcom_id])
+    @review = Review.find(params[:id])
+  end
+
+  def update
+    @review = Review.find(params[:id])
```

Comment:
You want to find sitcom first and then find the review using the associations.  So:
```ruby
@sitcom = Sitcom.find(params[:sitcom_id])
@review = @sitcom.reviews.find(params[:id])
```